### PR TITLE
Bring up CAN hardware on DIM

### PR DESCRIPTION
**Pull request by thekenu**
### Summary

Add just enough code so that we can send and receive CAN messages on DIM.

### Testing Done

Hooked up a CAN dongle to manually send/receive messages.

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [ ] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [ ] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: boards/DIM/Src/main.c [hunk 1]
```diff
@@ -68,14 +80,27 @@ static void MX_ADC2_Init(void);
...
 
 /* USER CODE BEGIN PFP */
 
 /* USER CODE END PFP */
 
 /* Private user code ---------------------------------------------------------*/
 /* USER CODE BEGIN 0 */
+static void CanRxQueueOverflowCallBack(size_t overflow_count)
+{
+    shared_assert(can_tx != NULL);
```
Comment (garethellis0) [inline boards/DIM/Src/main.c:93]: In keeping with the discussion we had recently about checking for NULL args, is this the sort of the place we need to do it? IMO it doesn't seem critical enough.

Perhaps add it to the auto-generated code for `CanTx` and `CanRx` so we don't have to worry about it?

File: boards/DIM/Src/main.c [hunk 2]
```diff
@@ -437,11 +480,35 @@ void RunTaskCanTx(void const *argument)
...
+
+    /* Infinite loop */
+    for (;;)
+    {
+        Io_CanTx_EnqueuePeriodicMsgs(
+            can_tx, osKernelSysTick() * portTICK_PERIOD_MS);
+        osDelayUntil(&PreviousWakeTime, period_ms);
+    }
+    /* USER CODE END StartTask1kHz */
+}
```
Comment (garethellis0) [inline boards/DIM/Src/main.c:510]: Please correct me if I'm wrong, but generally we have dedicated tasks for can RX and TX, no? Any particular reason we don't here?

File: boards/DIM/Src/main.c (hunk 1 above)
Comment (thekenu) [reply boards/DIM/Src/main.c:93]: Oh yeah I totally had the same thought while porting this code over. Let me clean this up for all the boards in this PR.

File: boards/DIM/Src/main.c (hunk 2 above)
Comment (thekenu) [reply boards/DIM/Src/main.c:510]: We do! 

```
void        RunTask100Hz(void const *argument);
void        RunTaskCanRx(void const *argument);
void        RunTaskCanTx(void const *argument);
void        StartTask1kHz(void const *argument);
```

The task order is just different on this board haha. I can go edit the `.ioc` so the CanRx and CanTx tasks are at the bottom like the other projects.

Comment (thekenu) [reply boards/DIM/Src/main.c:510]: Unless you're asking about something else? :thinking:

Comment (garethellis0) [reply boards/DIM/Src/main.c:510]: Nope, sorry, somehow just totally missed them here.

**Pull request merged**